// react-responsive-cards/dist/index.modern.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

var useStyles = makeStyles({
  root: {
    maxWidth: 455
  },
  media: {
    height: 300
  }
});
var Cards = function Cards(props) {
  var classes = useStyles();

  var renderCard = function renderCard(detail, index) {
    return /*#__PURE__*/React.createElement(Grid, {
      container: true,
      direction: "row",
      item: true,
      sm: 6,
      md: 4,
      lg: 3,
      alignItems: "center",
      justify: "center",
      key: index
    }, /*#__PURE__*/React.createElement(Card, {
      className: classes.root,
      key: detail.title,
      style: {
        margin: '6px'
      }
    }, /*#__PURE__*/React.createElement(CardActionArea, {
      onClick: detail.handleOnClick
    }, /*#__PURE__*/React.createElement(CardMedia, {
      "data-card-media": detail.title,
      className: classes.media,
      image: detail.image ? detail.image : 'https://via.placeholder.com/500x500.png?text=react-responsive-cards',
      title: detail.title
    }), /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Typography, {
      gutterBottom: true,
      variant: "h5",
      component: "h2"
    }, detail.title), /*#__PURE__*/React.createElement(Typography, {
      variant: "body2",
      color: "textSecondary",
      component: "p",
      style: {
        height: '50px'
      }
    }, detail.description))), detail.renderFooter));
  };

  return /*#__PURE__*/React.createElement(Grid, {
    container: true,
    justify: "center"
  }, props.details.map(function (detail, index) {
    return renderCard(detail, index);
  }));
};

export { Cards };
