// react-responsive-cards/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var styles = require('@material-ui/core/styles');
var Card = _interopDefault(require('@material-ui/core/Card'));
var CardActionArea = _interopDefault(require('@material-ui/core/CardActionArea'));
var CardContent = _interopDefault(require('@material-ui/core/CardContent'));
var CardMedia = _interopDefault(require('@material-ui/core/CardMedia'));
var Typography = _interopDefault(require('@material-ui/core/Typography'));
var Grid = _interopDefault(require('@material-ui/core/Grid'));

var useStyles = styles.makeStyles({
  root: {
    maxWidth: 455
  },
  media: {
    height: 300
  }
});
var Cards = function Cards(props) {
  var classes = useStyles();

  var renderCard = function renderCard(detail, index) {
    return /*#__PURE__*/React.createElement(Grid, {
      container: true,
      direction: "row",
      item: true,
      sm: 6,
      md: 4,
      lg: 3,
      alignItems: "center",
      justify: "center",
      key: index
    }, /*#__PURE__*/React.createElement(Card, {
      className: classes.root,
      key: detail.title,
      style: {
        margin: '6px'
      }
    }, /*#__PURE__*/React.createElement(CardActionArea, {
      onClick: detail.handleOnClick
    }, /*#__PURE__*/React.createElement(CardMedia, {
      "data-card-media": detail.title,
      className: classes.media,
      image: detail.image ? detail.image : 'https://via.placeholder.com/500x500.png?text=react-responsive-cards',
      title: detail.title
    }), /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(Typography, {
      gutterBottom: true,
      variant: "h5",
      component: "h2"
    }, detail.title), /*#__PURE__*/React.createElement(Typography, {
      variant: "body2",
      color: "textSecondary",
      component: "p",
      style: {
        height: '50px'
      }
    }, detail.description))), detail.renderFooter));
  };

  return /*#__PURE__*/React.createElement(Grid, {
    container: true,
    justify: "center"
  }, props.details.map(function (detail, index) {
    return renderCard(detail, index);
  }));
};

exports.Cards = Cards;
